src/spyglass/utils/dj_mixin.py

```diff
@@ -857,4 +857,9 @@ def delete(self, *args, **kwargs):
         """Delete master and part entries."""
         restriction = self.restriction or True  # for (tbl & restr).delete()
 
-        (self.master & restriction).delete(*args, **kwargs)
+        try:  # try restriction on master
+            restricted = self.master & restriction
+        except DataJointError:  # if error, assume restr of self
+            restricted = self & restriction
+
```

Comment:
Would this result in a loop in the except case? The delete on the mixin part would re-enter this function?

If the intention is to delete the master entry can we build a restriction appropriate for master and then call delete based on that?:
```python
master_restriction = [
    {k:v for k,v in key.items() if k in self.master.heading} 
    for key in (self & restriction).fetch("KEY")
]
restricted = self.master & master_restriction
```

Reply:
Thanks, @samuelbray32. I think my intention was to revert to the default `dj.Part` case outside of when we need it for propagation. I went back to #860 to make sure I could still propagate with this version - and, yes, it seems to still capture everything downstream.

![image](https://github.com/user-attachments/assets/25d45691-8d3d-4e5b-bf09-a2e2e209ece5)

Or are relevant `decoding` entries no longer present?